.github/scripts/main/main.sh

```diff
@@ -121,35 +163,94 @@ if [ "$OPAM_TEST" = "1" ]; then
-    git checkout -b "$OPAM_RT_BRANCH" "origin/$OPAM_RT_BRANCH"
-  fi
+  prepare_project "https://github.com/ocaml-opam/opam-rt" "opam-rt"
 
-  test -d _opam || opam switch create . --no-install --formula '"ocaml-system"'
-  eval $(opam env)
-  opam pin $GITHUB_WORKSPACE -yn --with-version to-test
   # opam lib pins defined in opam-rt are ignored as there is a local pin
   opam pin . -yn --ignore-pin-depends
-  opam install opam-rt --deps-only opam-devel.to-test
-  make || { opam reinstall opam-client -y; make; }
+  opam install opam-rt --deps-only opam-devel
+  opam exec -- make || { opam reinstall opam-client -y; opam exec -- make; }
   (set +x ; echo -en "::endgroup::opam-rt\r") 2>/dev/null
 fi
+
+test_project () {
+  url=$1
+  project=$2
+  ignore_depends=$3
+
+  ignore=""
+  if [ $ignore_depends -eq 1 ]; then
+    ignore="--ignore-pin-depends"
+  fi
+  (set +x; echo -en "::group::depends-$project\r") 2>/dev/null
+  prepare_project "$url" "$project"
+  set +e
+  opam pin "$url.git" -yn $ignore
+  for pkg_name in $(opam show . -f name); do
+
+    echo "Installing dependencies for $pkg_name"
+    opam install "$pkg_name" --deps-only
+    deps_code=$?
+    if [ $deps_code -ne 0 ]; then
+      echo "Dependency installation failed for $pkg_name"
+      DEPENDS_ERRORS="$DEPENDS_ERRORS $pkg_name"
+      set -e
+      (set +x ; echo -en "::endgroup::depends-$project\r") 2>/dev/null
+      return
+    fi
+
+    echo "Installing opam-client and $pkg_name"
+    opam install opam-client
+    opam install "$pkg_name"
+    code=$?
+    if [ $code -ne 0 ]; then
+      LIB_ERRORS="$LIB_ERRORS $project"
+    fi
+  done
+
+
+  set -e
+  (set +x ; echo -en "::endgroup::depends-$project\r") 2>/dev/null
+}
+
+if [ "$OPAM_DEPENDS" = "1" ]; then
+  DEPENDS_ERRORS=""
+  LIB_ERRORS=""
+
+  (set +x; echo -en "::group::depends\r") 2>/dev/null
+  eval $(grep '^VERSION=' shell/install.sh)
+  opam_libs=$(opam show . -f name 2>/dev/null)
+  depends_on=$(echo "$opam_libs" | sed "s/\$/.${VERSION}/" | paste -sd, -)
+
+  packages=$(opam list --or --depends-on "$depends_on" --columns name | \
+    tail -n +3)
+
+  for exclude in $opam_libs; do
+    packages=$(echo "$packages" | grep -v -x "$exclude")
+  done
+
+  for pkg in $packages; do
+    dev_repo=$(opam show "$pkg" -f dev-repo 2>/dev/null | head -n 1)
+
+    dev_repo=$(echo "$dev_repo" | sed -E 's/^"//;s/"$//;s/^git\+//;s/\.git$//')
+
+    if [[ -n "$dev_repo" ]]; then
+      test_project "$dev_repo" "$pkg" 0
+    fi
+  done
+
+  if [ -n "$DEPENDS_ERRORS" ]; then
+    echo -e "\e[31mErrors detected in dependencies of plugins $DEPENDS_ERRORS\e[0m";
+  fi
+
+  if [ -n "$LIB_ERRORS" ]; then
+    echo -e "\e[31mErrors detected in plugins $LIB_ERRORS\e[0m"
+    OCAMLVER=$(ocamlc -version)  
+    SAFE_VER=${OCAMLVER//./_}
+    echo "LIB_ERRORS_$SAFE_VER<<EOF" >> "$GITHUB_ENV"
+    echo "Failed to build : $LIB_ERRORS. OCaml version : $OCAMLVER. Commit $PR_REF_SHA. " >> "$GITHUB_ENV"
+    echo "EOF" >> "$GITHUB_ENV"
+  fi
+  
```

Comment:
For some projects that we manage (opam-publish fox ex), we want the job to fail. We could specify them via an environment variable in main.yml

Reply:
Done in `ci.ml`